contributingGuides/STYLE.md

```diff
@@ -477,20 +477,68 @@ if (ref.current && 'getBoundingClientRect' in ref.current) {
-const bar = report?.reportID ?? '0';
-
-report ? report.reportID : '0';
-report ? report.reportID : '';
+const accountID = report?.ownerAccountID ?? -1;
+const policyID = report?.policyID ?? '-1';
+const managerID = report ? report.managerID : 0;
 
 // GOOD
-const foo = report?.reportID ?? '-1';
+const accountID = report?.ownerAccountID ?? CONST.DEFAULT_NUMBER_ID;
+const policyID = report?.policyID;
+const managerID = report ? report.managerID : CONST.DEFAULT_NUMBER_ID;
+```
+
+Here are some common cases you may face when fixing your code to remove the old/bad default values.
+
+#### **Case 1**: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
+
+```diff
+-Report.getNewerActions(newestActionCurrentReport?.reportID ?? '-1', newestActionCurrentReport?.reportActionID ?? '-1');
++Report.getNewerActions(newestActionCurrentReport?.reportID, newestActionCurrentReport?.reportActionID);
+```
+
+> error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'. Type 'undefined' is not assignable to type 'string'.
+
+We need to change `Report.getNewerActions()` arguments to allow `undefined`. By doing that we could add a condition that return early if one of the parameters are falsy, preventing the code (which is expecting defined IDs) from executing.
+
+```diff
+-function getNewerActions(reportID: string, reportActionID: string) {
++function getNewerActions(reportID: string | undefined, reportActionID: string | undefined) {
++    if (!reportID || !reportActionID) {
++        return;
++    }
+```
+
+#### **Case 2**: Type 'undefined' cannot be used as an index type.
+
+```diff
+function MoneyRequestView({report, shouldShowAnimatedBackground, readonly = false, updatedTransaction, isFromReviewDuplicates = false}: MoneyRequestViewProps) {
+    const [parentReportActions] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${parentReportID}`, {
+        canEvict: false,
+    });
+-   const parentReportAction = parentReportActions?.[report?.parentReportActionID ?? '-1'];
++   const parentReportAction = parentReportActions?.[report?.parentReportActionID];
+```
+
+> error TS2538: Type 'undefined' cannot be used as an index type.
+
+This error is inside a component, so we can't simply use early return conditions here. Instead, we can use `String(report?.parentReportActionID)` to try to convert the value to `string`. If the value is `undefined`, the result string will be `'undefined'`, which will be used to find a record inside `parentReportActions` and, just like `-1`, would find nothing.
 
-report ? report.reportID : '-1';
+```diff
+function MoneyRequestView({report, shouldShowAnimatedBackground, readonly = false, updatedTransaction, isFromReviewDuplicates = false}: MoneyRequestViewProps) {
+-    const parentReportAction = parentReportActions?.[report?.parentReportActionID ?? '-1'];
++    const parentReportAction = parentReportActions?.[String(report?.parentReportActionID)];
```

Comment:
Agree that it's more aligned with the codebase, adjusted but used `undefined` instead of `null` to align more with Onyx data structure (Onyx doesn't return `null` values)

Reply:
In this case it's not an index of onyx, so no need to use `undefined` IMO. But NAB I think, what do you think @neil-marcellini

Reply:
Yes NAB, I'm fine with `undefined`.